arch/riscv/Kconfig

```diff
@@ -128,6 +128,46 @@ config GEN_IRQ_VECTOR_TABLE
 config NUM_IRQS
 	int
 
+config HAS_MONITOR_HART
+	bool "Does SOC have monitor hart with different capabilities to application harts"
+	default n
+	help
+	  SOC has monitor hart such as the e51 on MPFS and FU540 which has different
+	  capabilities to the other harts in the SOC. This may be important for SMP
+	  setups where the monitor hart may not be suitable for inclusion in the SMP
+	  operation.
+
+config MONITOR_HART_ID
+	int "Hart id for monitor hart"
+	default 0
+	help
+	  If the SOC has monitor hart such as the e51 on MPFS and FU540 this is
+	  the mhartid value for the monitor hart.
+
+config MONITOR_HART_SUPERVISOR
+	bool "Monitor hart supports supervisor mode"
+	default n
+	help
+	  If the SOC has monitor hart, this flag indicates if it supports supervisor
+	  mode. Many monitor harts such as the e51 on MPFS and FU540 do not support
+	  supervisor mode.
+
+config SUPERVISOR_MODE_CAPABLE
+	bool "Indicates that supervisor mode is available"
+	default y
```

Comment:
Why is `CONFIG_SUPERVISOR_MODE_CAPABLE` y-selected by default?

Reply:
Should indeed be n

Reply:
done